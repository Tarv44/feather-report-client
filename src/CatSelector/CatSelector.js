import React, {Component} from 'react';
import styles from './CatSelector.module.css';

export default class CatSelector extends Component {
    render() {
        const catButtons = this.props.categories.map((c, i) => {
            return <button onClick={e => this.props.handleFilter(c)} key={i}>{c}</button>
        })

        return (
            <section className={styles.section}>
                <h2>Categories</h2>
                <div className={styles.buttons}>
                    {catButtons}
                    <button onClick={e => this.props.handleFilter('')}>Clear Filter</button>
                </div>
            </section>
        )
    }
}