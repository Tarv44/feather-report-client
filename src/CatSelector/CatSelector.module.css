.buttons {
    display: flex;
    flex-flow: row wrap;
    justify-content: space-between;
    min-width: 100%;
}

.buttons button {
    margin: 0 0 20px;
    min-width: 48%;
    font-size: 16px;
    background-color: #26547C;
    color: #fff;
    border: solid black 2px;
    border-radius: 3px;
    padding: .3em .5em;
    box-shadow: 0 1px 10px 3px rgba(0,0,0,.1);
}

.section {
    min-width: 15%;
    padding: 0 10px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
}

.filter {
    color: red;
}

/* @media (min-width: 1280px) {

} */