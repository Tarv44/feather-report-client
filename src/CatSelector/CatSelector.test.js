import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import CatSelector from './CatSelector'
import { categories } from '../component.fixtures'


describe('----- CatSelector -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(<CatSelector categories={categories}/>, div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(<CatSelector categories={categories}/>)
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})