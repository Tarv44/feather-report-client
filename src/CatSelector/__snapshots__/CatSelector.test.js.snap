// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- CatSelector ----- renders the UI as expected 1`] = `
<section
  className="section"
>
  <h2>
    Categories
  </h2>
  <div
    className="buttons"
  >
    <button
      id={null}
      onClick={[Function]}
    >
      Knives
    </button>
    <button
      id={null}
      onClick={[Function]}
    >
      Blenders
    </button>
    <button
      id={null}
      onClick={[Function]}
    >
      Toasters
    </button>
    <button
      onClick={[Function]}
    >
      All Categories
    </button>
  </div>
</section>
`;
